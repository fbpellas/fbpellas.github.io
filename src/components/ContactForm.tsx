import * as React from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import { EMAIL } from '../constants';

export interface ContactFormProps { }

const ContactForm: React.FC<ContactFormProps> = (_props) => {
  const [emailBody, setEmailBody] = React.useState('');
  const [emailSubject, setEmailSubject] = React.useState('');
  const buttonRequest = `mailto:${EMAIL}?subject=${encodeURIComponent(emailSubject)}&body=${encodeURIComponent(emailBody)}`;

  return <Form className="form">
    <Form.Group controlId="exampleForm.ControlTextarea1">
      <Form.Label>Email form</Form.Label>
      <Form.Text className="text-muted">{`Or send me an email directly at ${EMAIL}`}</Form.Text>
      <br />
      <Form.Label>Subject</Form.Label>
      <Form.Control
        onChange={(e) => {
          setEmailSubject(e.target.value);
        }}
        type="text"
        placeholder="Subject"
      />
      <br />
      <Form.Label>Body</Form.Label>
      <Form.Control
        onChange={(e) => {
          setEmailBody(e.target.value);
        }}
        as="textarea"
        rows={10}
        placeholder="Your message"
      />
    </Form.Group>
    <br />
    <Button
      onClick={() => {
        window.open(
          buttonRequest
        );
      }}
      variant="warning"
      type="submit"
    >
      Send
    </Button>
  </Form>
}

export { ContactForm }