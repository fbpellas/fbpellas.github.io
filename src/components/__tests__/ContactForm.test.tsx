import React from 'react';
import { render } from '@testing-library/react';
import { ContactForm } from '../ContactForm';

test('ContactForm', () => {
  const { container } = render(<ContactForm />);
  expect(container).toMatchSnapshot();
});
