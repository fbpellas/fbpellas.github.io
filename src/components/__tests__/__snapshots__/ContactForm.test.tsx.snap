// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm 1`] = `
<div>
  <form
    class="form"
  >
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="exampleForm.ControlTextarea1"
      >
        Email form
      </label>
      <small
        class="text-muted form-text"
      >
        Or send me an email directly at fbpellas@usfca.edu
      </small>
      <br />
      <label
        class="form-label"
        for="exampleForm.ControlTextarea1"
      >
        Subject
      </label>
      <input
        class="form-control"
        id="exampleForm.ControlTextarea1"
        placeholder="Subject"
        type="text"
      />
      <br />
      <label
        class="form-label"
        for="exampleForm.ControlTextarea1"
      >
        Body
      </label>
      <textarea
        class="form-control"
        id="exampleForm.ControlTextarea1"
        placeholder="Your message"
        rows="10"
      />
    </div>
    <br />
    <button
      class="btn btn-warning"
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;
